api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,35 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  const token = req.rawHeaders.find(element => element.length > 50);
+  if (!token) {
+    throw new ClientError(401, 'authentication required');
+  }
+  try {
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (err) {
+    console.error(err);
+  }
```

Comment:
If we catch and error, we need to pass it to `next()` so that our error middleware can deal with it.